index.js

```diff
@@ -13,16 +14,22 @@ module.exports.file = function file$promise() {
 
 module.exports.withFile = function withFile(fn) {
   var cleanup;
-  return module.exports.file.apply(tmp).then(function context(o) { 
+
+  var params = Array.prototype.slice.call(arguments);
+  params.shift();
```

Comment:
Are you sure this should be `shift`? Also, I'm wondering if it's not enough to do `Array.from` rather than Array.slice.call here.